src/mpid/ch4/netmod/ofi/ofi_vci.c

```diff
@@ -30,6 +30,13 @@ int MPIDI_OFI_comm_set_vcis(MPIR_Comm * comm, int num_vcis, int *all_num_vcis)
         MPIDI_OFI_am_post_recv(vci, 0);
     }
 
+    if (MPIR_CVAR_DEBUG_SUMMARY && MPIR_Process.rank == 0) {
+        fprintf(stdout, "==== OFI dynamic settings ====\n");
+        fprintf(stdout, "num_vcis: %d\n", MPIDI_OFI_global.num_vcis);
+        fprintf(stdout, "num_nics: %d\n", MPIDI_OFI_global.num_nics);
+        fprintf(stdout, "======================================\n");
+    }
```

Comment:
Does this change mean that users may see this output at a time other than MPI_INIT? Possibly multiple times?

Reply:
`MPIR_Process.rank ==0` should be `comm->rank == 0`. 

Yes, users may see this output multiple times at run time if we expose it via e.g. `MPIX_Comm_set_vcis` and if users call it multiple times (it has to be non-overlapping comms though). Currently, it is still part of comm world initialization. 

We need a way to feedback to user on whether multiple vcis are in use. I would argue that `MPIX_Comm_set_vcis` will only be used at set up time in the similar nature as `MPI_Init` from user's point of view. In an MPI session model, there isn't comm world initialization, creating a comm and setting up the comm is the initialization for an MPI environment. In a sense, we are splitting `MPI_Init` so users have more control.

Reply:
> Yes, users may see this output multiple times at run time if we expose it via e.g. `MPIX_Comm_set_vcis` and if users call it multiple times (it has to be non-overlapping comms though). Currently, it is still part of comm world initialization.

Got it. I guess this won't have impact and any extension that uses it will be opt-in with a chance for feedback.